src/target/adiv5.c

```diff
@@ -341,10 +346,11 @@ static bool adiv5_component_probe(ADIv5_AP_t *ap, uint32_t addr, bool do_test, i
 		}
 		DEBUG("%sROM: Table END\n", indent);
 	} else {
-		/* Check if the component was designed by ARM, we currently do not support,
-		 * any components by other designers.
+		/* Check if the component was designed by ARM or Atmel DSU, we currently
+		 * do not support any components by other designers.
 		 */
-		if ((pidr & ~(PIDR_REV_MASK | PIDR_PN_MASK)) != PIDR_ARM_BITS) {
+		uint64_t pidr_jep_bits = (pidr & ~(PIDR_REV_MASK | PIDR_PN_MASK));
+		if (pidr_jep_bits != PIDR_ARM_BITS && pidr_jep_bits != PIDR_ATMEL_DSU_BITS) {
 			DEBUG("%s0x%"PRIx32": 0x%"PRIx64" <- does not match ARM JEP-106\n",
 				  indent, addr, pidr);
 			return false;
```

Comment:
Returning false here, means that cortexm_prepare is called here: https://github.com/UweBonnes/blackmagic/blob/3cac3d9cdb7e0dc17f75134cc7a674e81af94b92/src/target/adiv5.c#L514-L517

and as cortexm_prepare returns false, probing fails.

Reply:
The protected SAMD should not expose the CortexM DAP, so this path should not be taken. Can you investigate?